### PR TITLE
LwM2M: separate write operation from write-attributes op

### PR DESCRIPTION
**Pull request by rbtchc**
Content-format is used to determine the type of the PUT/POST request.
Therefore, it's incorrect to assign default when the caller does not include one in the request.

Changes are listed below
1. Define LWM2M_FORMAT_NONE=65535 to indicate the format is missing.
    The 65000~65535 is reserved for experiments and should be safe for the purpose.
1.  Check content-type at PUT method to setup write/write-attrs operation accordingly.
1.  Add reporting write-attrs as not implemented to the caller.

One non-related patch is to re-write coap_options_to_path() to save some code size.

### CODE REVIEW

File: subsys/net/lib/lwm2m/lwm2m_engine.c [hunk 1]
```diff
@@ -702,32 +702,16 @@ static u16_t atou16(u8_t *buf, u16_t buflen, u16_t *len)
 static int coap_options_to_path(struct coap_option *opt, int options_count,
 				struct lwm2m_obj_path *path)
 {
-	u16_t len;
+	u16_t len, *id[4] = {&path->obj_id, &path->obj_inst_id,
```
Comment (mike-scott) [inline subsys/net/lib/lwm2m/lwm2m_engine.c]: can we add a space after the { and before the } on the next line?

File: subsys/net/lib/lwm2m/lwm2m_engine.c [hunk 2]
```diff
@@ -2512,7 +2509,9 @@ static int handle_request(struct coap_packet *request,
 		break;
 
 	case COAP_METHOD_PUT:
-		context.operation = LWM2M_OP_WRITE;
+		/* write attributes if content-format is absent */
+		context.operation = format == LWM2M_FORMAT_NONE ?
```
Comment (mike-scott) [inline subsys/net/lib/lwm2m/lwm2m_engine.c]: I know it's not necessary but can we add parentheses to clarify the operations here.

**Review by mike-scott — COMMENTED** [review]
minor nits, but overall looks good.

**Comment by rbtchc** [comment]
Update according to the review
Diff: https://hastebin.com/heveretiwu.cpp

**Review by mike-scott — APPROVED** [review]
LGTM

**Comment by mike-scott** [comment]
Adding @jukkar and @nashif

**Pull request merged**